src/drivers/uavcannode/UavcanNode.cpp

```diff
@@ -824,6 +833,21 @@ extern "C" int uavcannode_start(int argc, char *argv[])
 		}
 	}
 
+	// Read the static node ID parameter and use it if no valid node_id from shared memory
+	int32_t cannode_node_id = 0;
+	param_get(param_find("CANNODE_NODE_ID"), &cannode_node_id);
+
+	// Check if the static node ID is in range
+	if (cannode_node_id < 0 || cannode_node_id > uavcan::NodeID::Max) {
+		PX4_ERR("Invalid static node ID %ld, using dynamic allocation", cannode_node_id);
+		cannode_node_id = 0;
+	}
+
+	// Assign the static node ID if no dynamic allocation is used. Do wen't override a valid node ID from the bootloader?
+	if (node_id == 0 && cannode_node_id > 0 && cannode_node_id <= uavcan::NodeID::Max) {
```

Comment:
This logic can actually be simplified by combining with the above range check. The way this works is if a bootloader does not assign an ID (whether statically or through DNA) then the CANNODE_NODE_ID will be used. The default cannode bootloader behavior is to **not** assign an ID and **not** perform allocation. So the only time CANNODE_NODE_ID will not be used is if the bootloader assigns a node ID.